framework/src/modules/interoperability/mainchain/commands/message_recovery_initialization.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { codec } from '@liskhq/lisk-codec';
+import { utils } from '@liskhq/lisk-cryptography';
+import { SparseMerkleTree } from '@liskhq/lisk-db';
+import {
+	CommandExecuteContext,
+	CommandVerifyContext,
+	VerificationResult,
+	VerifyStatus,
+} from '../../../../state_machine/types';
+import { ChannelData } from '../../types';
+import { BaseInteroperabilityCommand } from '../../base_interoperability_command';
+import { MainchainInteroperabilityInternalMethod } from '../internal_method';
+import { getMainchainID } from '../../utils';
+import { EMPTY_BYTES } from '../../constants';
+import { messageRecoveryInitializationParamsSchema } from '../../schemas';
+import { OwnChainAccountStore } from '../../stores/own_chain_account';
+import { ChainAccountStore } from '../../stores/chain_account';
+import { TerminatedStateStore } from '../../stores/terminated_state';
+import { ChannelDataStore, channelSchema } from '../../stores/channel_data';
+
+export interface MessageRecoveryInitializationParams {
+	chainID: Buffer;
+	channel: Buffer;
+	bitmap: Buffer;
+	siblingHashes: Buffer[];
+}
+
+export class MessageRecoveryInitializationCommand extends BaseInteroperabilityCommand<MainchainInteroperabilityInternalMethod> {
+	public schema = messageRecoveryInitializationParamsSchema;
+
+	public async verify(
+		context: CommandVerifyContext<MessageRecoveryInitializationParams>,
+	): Promise<VerificationResult> {
+		const { params } = context;
+
+		const ownchain = await this.stores.get(OwnChainAccountStore).get(context, EMPTY_BYTES);
+		const mainchainID = getMainchainID(ownchain.chainID);
+		if (params.chainID.equals(mainchainID) || params.chainID.equals(ownchain.chainID)) {
+			throw new Error('Chain ID is not valid.');
+		}
+
+		const chainAccountExist = await this.stores.get(ChainAccountStore).has(context, params.chainID);
+		if (!chainAccountExist) {
+			throw new Error('Chain account is not registered.');
+		}
+
+		const terminatedAccount = await this.stores
+			.get(TerminatedStateStore)
+			.get(context, params.chainID);
+		if (!terminatedAccount.initialized) {
+			throw new Error('Chain is not terminated.');
+		}
+
+		const queryKey = Buffer.concat([
```

Comment:
`STORE_PREFIX_INTEROPERABILITY ` is missing in the queryKey?

Reply:
key contains both module and store key